conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -519,12 +528,70 @@ private CodeBlock generateParameterCodeBlock(Stream<ArgumentDefinition> params,
                                         arg.getArgName().get(),
                                         createConstructorForTypeWithReference(arg.getType(), rawVarName,
                                                 typeDefinitions, typeMapper)
-                                )
+                                ),
+                                generateStoreParamInExchange(arg)
                         );
                     }
                 }).collect(Collectors.toList()));
     }
 
+    private Optional<String> getStoreMethodAndParamName(ArgumentDefinition arg) {
+        boolean isSafe = arg.getMarkers().stream().anyMatch(type ->
+                type.accept(
+                        new DefaultTypeVisitor<Boolean>() {
+                            @Override
+                            public Boolean visitExternal(ExternalReference value) {
+                                return value.getExternalReference().getName().equals("Safe");
+                            }
+
+                            @Override
+                            public Boolean visitDefault() {
+                                return false;
+                            }
+                        }));
+        if (isSafe) {
+            return arg.getParamType().accept(new ParameterType.Visitor<Optional<String>>() {
+                @Override
+                public Optional<String> visitHeader(HeaderParameterType value) {
+                    return Optional.of(value.getParamId().get());
```

Comment:
We probably should add checks here to prevent logging non-whitelisted headers even if they are incorrectly marked `Safe`.